pkg/agent/cniserver/server.go

```diff
@@ -527,6 +529,20 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	}
 	cniVersion := cniConfig.CNIVersion
 	cniResult, _ := result.Result.GetAsVersion(cniVersion)
+	status, err := netdefutils.CreateNetworkStatus(cniResult, cniConfig.Name, true, nil)
+	if err != nil {
+		klog.ErrorS(err, "Create NetworkStatus failed", "Pod", klog.KRef(podNamespace, podName))
+	} else {
+		pod, err := s.kubeClient.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
+		if err == nil {
+			netStatus := []netdefv1.NetworkStatus{*status}
+			if err := netdefutils.SetNetworkStatus(s.kubeClient, pod, netStatus); err != nil {
+				klog.ErrorS(err, "Pod network status annotation update failed", "Pod", klog.KObj(pod))
+			} else {
+				klog.V(2).InfoS("Pod network status annotation updated", "Pod", klog.KObj(pod), "NetworkStatus", netStatus)
+			}
+		}
+	}
```

Comment:
I believe we only need to add the status annotation for the Pod which is enabled with SecondaryNetwork annotation, probably check the annotation `k8s.v1.cni.cncf.io/networks` first.
cc @tnqn @antoninbas

Reply:
yes, I was going to comment that doing this unconditionally for all Pods, regardless of whether secondary network interfaces are present and regardless of the state of the `SecondaryNetwork` feature gate is probably wrong.

Reply:
@wenqiq , reminder, it seems comments are not addressed.

Reply:
Updated.